src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -230,6 +231,8 @@ await listener.AfterValidationAsync(context, validationResult)
                     };
                 }
 
+                executionOccurred = true;
```

Comment:
The positioning here seems to make sense.  The document listeners could throw an exception, but they could be considered part of the execution stage....

Reply:
Yes. Listeners "belong" to execution.